### PR TITLE
Added derivePassword implementing Master Password

### PR DESCRIPTION
**Pull request by fwiesel**
derivePassword derives a password from a given master password and a set of other parameters.
The tests are variants of the input comparing it to the reference implementation.

### CODE REVIEW

**Comment by fwiesel** [comment]
This is especially helpful in helm style deployments, where you might avoid re-using the same password over the landscape directly.

**Comment by databus23** [comment]
ping @technosophos. Any comments on this? We would find this quite useful to avoid having to set 20 passwords in out openstack chart value files while avoiding rolling out the same password to all services.

File: functions.go [hunk 1]
```diff
@@ -172,21 +172,27 @@ Crypto Functions:
 	- genPrivateKey: Generate a private key for the given cryptosystem. If no
 	  argument is supplied, by default it will generate a private key using
 	  the RSA algorithm. Accepted values are `rsa`, `dsa`, and `ecdsa`.
-
+	- derivePassword: Derive a password from the given parameters according to the ["Master Password" algorithm](http://masterpasswordapp.com/algorithm.html)
```
Comment (technosophos) [inline functions.go]: The markdown on this line won't get resolved by the godoc tool. Can you just remove the square brackets?

**Comment by technosophos** [comment]
Yeah, I really like this. I went back and forth on whether it was better suited for Helm internals, but I think it is beneficial for a broader range of users. If you wouldn't mind that one minor doc cleanup, @fwiesel , I will merge this. I am already planning a Sprig 2.9 release, so this will go into that release.

**Comment by fwiesel** [comment]
@technosophos  I've removed the brackets. Also the example was misleading, I've changed that too.

Thanks for the review.

**Pull request merged**